sdk/src/sdk/pubsub/MqttPubSubClient.cpp

```diff
@@ -85,6 +85,34 @@ class MqttPubSubClient : public IPubSubClient, private mqtt::callback {
 
         m_client.connect(m_connectOptions)->wait();
     }
+
+    void reconnect(int timeout_ms) override {
+        constexpr int MAX_TIMEOUT_MS = 30000;
+        logger().info("Attempting to reconnect to MQTT broker");
+        if (timeout_ms <= 0) {
+            logger().error("Invalid timeout value: {} ms. Must be positive.",
+                           timeout_ms);
+            return;
+          }
+
+        if (timeout_ms > MAX_TIMEOUT_MS) {
+          logger().warn("Timeout capped to {} ms (requested: {} ms)",
+                        MAX_TIMEOUT_MS, timeout_ms);
+          timeout_ms = MAX_TIMEOUT_MS;
+        }
+
```

Comment:
Don't know if this limitation is required. I would leave it to the user to decide about this.